x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -70,3 +186,68 @@ func createPoolRestriction(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
 		return hasTwoCoins && hasPoolCreationFee
 	}
 }
+
+// getRandCLPool gets a concentrated liquidity pool with its pool denoms.
+func getRandCLPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (cltypes.ConcentratedPoolExtension, []string, error) {
+	// get all pools
+	clPools, err := k.GetAllPools(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	numPools := len(clPools)
+	if numPools == 0 {
+		return nil, nil, fmt.Errorf("no pools created")
+	}
+
+	randConcentratedPool := clPools[rand.Intn(numPools)]
```

Comment:
So this seems to be accessed from `math/rand`. Can you please remove that import and ensure that we are accessing all randomness from the simulator-specific rand manager?